protocol/x/rewards/keeper/keeper.go

```diff
@@ -126,26 +128,45 @@ func (k Keeper) GetRewardShare(
 //     by quote quantums of the fill.
 func (k Keeper) AddRewardSharesForFill(
 	ctx sdk.Context,
-	takerAddress string,
-	makerAddress string,
-	bigFillQuoteQuantums *big.Int,
-	bigTakerFeeQuoteQuantums *big.Int,
-	bigMakerFeeQuoteQuantums *big.Int,
+	fill clobtypes.FillForProcess,
+	revshares []revsharetypes.RevShare,
 ) {
 	// Process reward weight for taker.
 	lowestMakerFee := k.feeTiersKeeper.GetLowestMakerFee(ctx)
 	maxMakerRebatePpm := lib.Min(int32(0), lowestMakerFee)
+
+	totalNetFeeRevSharePpm := uint32(0)
+	affiliateRevShareQuoteQuantums := big.NewInt(0)
+	for _, revshare := range revshares {
+		if revshare.RevShareFeeSource == revsharetypes.REV_SHARE_FEE_SOURCE_NET_FEE {
+			totalNetFeeRevSharePpm += revshare.RevSharePpm
+		}
+		if revshare.RevShareType == revsharetypes.REV_SHARE_TYPE_AFFILIATE {
```

Comment:
Think we are conflating two concepts here. We should only care about `FeeSource` in this logic (i.e. Reward keeper just cares about taker fee share instead of whether it goes to an affiliates)

Reply:
Also think we can avoid this loop by return additional fields [here](https://github.com/dydxprotocol/v4-chain/pull/2251#discussion_r1761295637)

Reply:
Good callout